File: src/main/java/com/moira/itda/global/entity/GachaItem.java
package com.moira.itda.global.entity;

import com.moira.itda.domain.series.dto.request.SeriesItemAddRequest;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.ZonedDateTime;

@Builder
@Getter
@Setter
public class GachaItem {
    private Long id;
    private String seriesId;
    private String name;
    private ZonedDateTime createdAt;

    public static GachaItem fromSeriesItemAddRequest(String seriesId, SeriesItemAddRequest request) {
        return GachaItem.builder()
                .seriesId(seriesId)
                .name(request.name())
                .createdAt(ZonedDateTime.now())
                .build();
    }
}

File: src/main/kotlin/com/moira/itda/global/entity/GachaItem.kt
package com.moira.itda.global.entity

import com.moira.itda.domain.admin.gacha.dto.request.AdminGachaItemAddRequest
import java.time.ZonedDateTime

data class GachaItem(
    val id: Long?,
    val gachaId: String,
    val name: String,
    val fileId: String?,
    val createdAt: ZonedDateTime,
    val updatedAt: ZonedDateTime
) {
    companion object {
        fun from(gachaId: String, request: AdminGachaItemAddRequest): GachaItem {
            return GachaItem(
                id = null,
                gachaId = gachaId,
                name = request.name,
                fileId = null, // TODO: 추후 가챠 아이템에도 이미지 파일 저장
                createdAt = ZonedDateTime.now(),
                updatedAt = ZonedDateTime.now()
            )
        }
    }
}